src/modules/rt_events/RT_E_TRAIN_fbt.cpp

```diff
@@ -59,11 +59,36 @@ namespace {
       .mSocketNames = {},
       .mPlugNames = {},
   };
+
+  const auto cEventConnections = std::array<SCFB_FBConnectionData, 7>{{
+      {STRID(RT_E_CYCLE), STRID(EO), STRID(E_CTU), STRID(CU)},
+      {CStringDictionary::scmInvalidStringId, STRID(START), STRID(E_CTU), STRID(R)},
+      {CStringDictionary::scmInvalidStringId, STRID(STOP), STRID(RT_E_CYCLE), STRID(STOP)},
+      {STRID(E_CTU), STRID(RO), STRID(RT_E_CYCLE), STRID(START)},
+      {STRID(E_CTU), STRID(CUO), STRID(E_SWITCH), STRID(EI)},
+      {STRID(E_SWITCH), STRID(EO0), CStringDictionary::scmInvalidStringId, STRID(EO)},
+      {STRID(E_SWITCH), STRID(EO1), STRID(RT_E_CYCLE), STRID(STOP)},
+  }};
```

Comment:
I suggest using `std::to_array`[^1], which allows deducing the length:
```suggestion
  const auto cEventConnections = std::to_array<SCFB_FBConnectionData>({
      {STRID(RT_E_CYCLE), STRID(EO), STRID(E_CTU), STRID(CU)},
      {CStringDictionary::scmInvalidStringId, STRID(START), STRID(E_CTU), STRID(R)},
      {CStringDictionary::scmInvalidStringId, STRID(STOP), STRID(RT_E_CYCLE), STRID(STOP)},
      {STRID(E_CTU), STRID(RO), STRID(RT_E_CYCLE), STRID(START)},
      {STRID(E_CTU), STRID(CUO), STRID(E_SWITCH), STRID(EI)},
      {STRID(E_SWITCH), STRID(EO0), CStringDictionary::scmInvalidStringId, STRID(EO)},
      {STRID(E_SWITCH), STRID(EO1), STRID(RT_E_CYCLE), STRID(STOP)},
  });
```

[^1]: https://en.cppreference.com/w/cpp/container/array/to_array.html

Reply:
oh yes this is much better. Changed.